### ISSUE TITLE
Incomplete scan report when using local paths in package.json

### ISSUE THREAD

**Issue by ageorgez** [TrustSource/ts-node-client#28]
ts-node-client 3.2.2
node 18.18.2
npm 9.8.1

In our project, we're using two local paths in the **package.json** to be able to generate some packages directly in the node_modules:

`"@core/styles": "file:libs/shared/design/core-styles/src/styles",`
...
`"eslint-plugin-custom-rules": "file:tools/eslint/eslint-custom-rules"`

After running `npm install` the **package-lock.json** will have the following entries for `"@core/styles"` and `"eslint-plugin-custom-rules"`:

`"node_modules/@core/styles": {
            "resolved": "libs/shared/design/core-styles/src/styles",
            "link": true
        },` 
  ...
`"libs/shared/design/core-styles/src/styles": {},`
...
`"node_modules/eslint-plugin-custom-rules": {
            "resolved": "tools/eslint/eslint-custom-rules",
            "link": true
        },`
...
`"tools/eslint/eslint-custom-rules": {
            "name": "eslint-plugin-custom-rules",
            "version": "1.0.0",
            "dev": true,
            "license": "ISC"
        }`

Because of these 4 entries when running the scan with ts-node-client we get an incomplete report.
Removing the entries above we get a complete report.

**Comment by olexandrbig** [comment]
Hello @ageorgez , your request is under investigation

**Comment by olexandrbig** [comment]
Tested under:
windows 10
ts-node-client 3.2.2
node 18.18.2
npm 9.8.1

Sample of problem:

_package.json_
```
{
  "name": "ts-test",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "scan": "ts-node-client --debug --saveAs test --saveAsFormat scan"
  },
  "dependencies": {
    "@core/styles": "file:libs/shared/design/core-styles/src/styles",
    "eslint-plugin-custom-rules": "file:tools/eslint/eslint-custom-rules",
    "ts-node-client": "3.2.2"
  }
}

```
_scan.json_
```
{
  "project": "API1",
  "module": "ts-test",
  "moduleId": "npm:ts-test",
  "dependencies": [
    {
      "name": "ts-test",
      "key": "npm:ts-test",
      "private": true,
      "licenses": [],
      "repoUrl": null,
      "dependencies": [
        {
          "name": "@core/styles",
          "key": "npm:@core/styles",
          "private": false,
          "licenses": [],
          "repoUrl": null,
          "dependencies": [],
          "versions": [
            "file:libs/shared/design/core-styles/src/styles"
          ]
        },
        {
          "name": "eslint-plugin-custom-rules",
          "key": "npm:eslint-plugin-custom-rules",
          "private": false,
          "licenses": [],
          "repoUrl": null,
          "dependencies": [],
          "versions": [
            "file:tools/eslint/eslint-custom-rules"
          ]
        },
        {
          "name": "ts-node-client",
          "key": "npm:ts-node-client",
          "private": false,
          "licenses": [],
          "repoUrl": null,
          "dependencies": [],
          "versions": [
            "3.2.2"
          ]
        }
      ],
      "versions": [
        "1.0.0"
      ]
    }
  ]
}
```

Package.lock file is missing version for local dependencies. Result of parsing is invalid and fallback to package.json is used

Update will be available under ts-node-client@3.2.3
